src/Illuminate/Http/Resources/Json/JsonResource.php

```diff
@@ -46,13 +50,27 @@ class JsonResource implements ArrayAccess, JsonSerializable, Responsable, UrlRou
      */
     public static $wrap = 'data';
 
+    /**
+     * The default namespace where factories reside.
+     *
+     * @var string
+     */
+    protected static $namespace = 'App\\Http\\Resources\\';
```

Comment:
I don't think the http component can be allowed to know what `App` is. That is a job for Foundation, or the actual app.